tianshou/utils/net/discrete.py

```diff
@@ -163,3 +163,37 @@ def forward(
         x = x.view(-1, self.num_atoms).softmax(dim=-1)
         x = x.view(-1, np.prod(self.action_shape), self.num_atoms)
         return x, state
+
+
+class QRDQN(DQN):
+    """Reference: Distributional Reinforcement Learning with Quantile Regression.
+
+    For advanced usage (how to customize the network), please refer to
+    :ref:`build_the_network`.
+    """
+
+    def __init__(
+        self,
+        c: int,
+        h: int,
+        w: int,
+        action_shape: Sequence[int],
+        num_quantiles: int = 200,
+        device: Union[str, int, torch.device] = "cpu",
+    ) -> None:
+        super().__init__(c, h, w,
+                         [np.prod(action_shape) * num_quantiles], device)
+        self.action_shape = action_shape
+        self.num_quantiles = num_quantiles
+
+    def forward(
+        self,
+        x: Union[np.ndarray, torch.Tensor],
+        state: Optional[Any] = None,
+        info: Dict[str, Any] = {},
+    ) -> Tuple[torch.Tensor, Any]:
+        r"""Mapping: x -> Z(x, \*)."""
+        x, state = super().forward(x)
+        x = x.view(-1, self.num_quantiles)
+        x = x.view(-1, np.prod(self.action_shape), self.num_quantiles)
```

Comment:
why two view operations?

Reply:
This is my fault. One view operation is enough.